### DISCUSSION TITLE
Support profiling queries via tracing

### DISCUSSION THREAD

**Discussion by iskakaushik** [lquerel/gcp-bigquery-client#33] [Ideas]
Adding support for tracing via https://docs.rs/tracing/latest/tracing/ would be useful. Given that most of the requests are going through [reqwest](https://docs.rs/reqwest/latest/reqwest/), seems like https://github.com/TrueLayer/reqwest-middleware might be a good candidate.

Let me know if this is something that you think will be generally useful and I'm happy to submit a PR for this. This can be behind a [cargo feature](https://doc.rust-lang.org/cargo/reference/features.html), maybe `profile`, that way it can be enabled only when profiling.

Thanks for the awesome crate.

**Comment by lquerel** [comment]
Thanks for the suggestion and I agree that adding a tracing capability would be helpful. Please submit a PR with an implementation of your proposal. Regards, Laurent
